r/tests/testthat/test-Array.R

```diff
@@ -839,3 +840,23 @@ test_that("Array to C-interface", {
   delete_arrow_schema(schema_ptr)
   delete_arrow_array(array_ptr)
 })
+
+test_that("posixlt", {
+  skip("Arrays don't keep metadata about class type")
```

Comment:
This is true both before and after this PR. I'm happy to take these additional tests out since they don't work today and won't work without additional work.

When I was chatting with Joris about this, he suggested that we might want to lean on using an extension type of struct here and that should give us enough info to reconstruct a posixlt from an Array of them directly.